### ISSUE TITLE
Terraform validate `--retry-once-with-cleanup=true` not cleaning up all `.terraform/providers` issues

### ISSUE THREAD

**Issue by brettcurtis** [antonbabenko/pre-commit-terraform#561]
### Describe the bug

I'm not exactly sure how I end up in this position, but in some cases, I see:

```none
Error: registry.terraform.io/hashicorp/google: there is no package for registry.terraform.io/hashicorp/google 4.79.0 cached in .terraform/providers

Validation failed: global/infra
{
  "format_version": "1.0",
  "valid": false,
  "error_count": 1,
  "warning_count": 0,
  "diagnostics": [
    {
      "severity": "error",
      "summary": "registry.terraform.io/hashicorp/google: there is no package for registry.terraform.io/hashicorp/google 4.79.0 cached in .terraform/providers",
      "detail": ""
    }
  ]
}
```

Should `--retry-once-with-cleanup=true` look for this message and clean up as well?

### How can we reproduce it?

To reproduce you can simply delete the `.terraform/providers/registry.terraform.io/hashicorp/google/4.79.0` directory.

**Comment by MaxymVlasov** [comment]
Please
1. Provide reproducible instructions from scratch.
2. Provide other info ignored by you during issue creation. Fee free find it in [issue template](https://github.com/antonbabenko/pre-commit-terraform/blob/master/.github/ISSUE_TEMPLATE/bug_report_local_install.md?plain=1)
3. Check that you set the `TF_PLUGIN_CACHE_DIR` variable because the error looks like you don't use it and are lucky to have networking issues, or use TF 1.4+ but without lockfiles.

**Comment by brettcurtis** [comment]
Sure here are the steps to repro:

```none
gh repo clone osinfra-io/google-cloud-networking
```

```none
cd google-cloud-networking
```

```none
pre-commit run -a
```

```none
rm -rf global/infra/.terraform/providers/registry.terraform.io/hashicorp/google/4.80.0 
```

```none
pre-commit run -a
```

### Environment information

```bash
Linux SURFACE 5.15.90.1-microsoft-standard-WSL2
```

```bash
GNU bash, version 5.2.15(1)-release (x86_64-pc-linux-gnu)
pre-commit 3.3.3
Terraform v1.5.6
Python 3.11.5
Python 3.11.5
checkov checkov SKIPPED
terraform-docs version v0.16.0 linux/amd64
terragrunt SKIPPED
terrascan terrascan SKIPPED
tflint SKIPPED
tfsec tfsec SKIPPED
tfupdate tfupdate SKIPPED
hcledit hcledit SKIPPED
```

3: I'll check out the cache stuff for local development, I do not use it now and it seems it could help.

**Comment by MaxymVlasov** [comment]
Bug confirmed

**Comment by antonbabenko** [comment]
This issue has been resolved in version 1.83.2 :tada:
